addons/l10n_mx_pos/models/pos_order.py

```diff
@@ -0,0 +1,65 @@
+from odoo import models, _
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    def action_invoice_order(self):
+        self.env.context.update({
+            'state' : 'draft'
+        })
+        return {
+            'type': 'ir.actions.act_window',
+            'name': 'Select a Customer',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'view_id' :  self.env.ref('l10n_mx_pos.global_customer_select_form').id,
+            'res_model' : 'pos.order.select',
+            'target' : 'new',
+            'flags': {'form': {'action_buttons': False}},
+            'context' : self.env.context
+        }
+
+    def _generate_multi_pos_order_invoice(self):
+
+        configs = {}
+        moves = self.env['account.move']
+
+        for order in self:
+            if order.config_id in configs:
+                configs[order.config_id] += order
+            else:
+                configs[order.config_id] = order
+
+        for config, orders in configs.items():
+            move_vals = {
+                'journal_id' : config.journal_id.id,
+                'partner_id' : orders[0].partner_id.id,
+                'move_type': 'out_invoice' if orders[0].amount_total >= 0 else 'out_refund',
+                'currency_id' : config.currency_id.id,
+                'fiscal_position_id' : config.default_fiscal_position_id.id,
+                'invoice_line_ids' : []
+            }
+            for order in orders:
+                move_vals['invoice_line_ids'].extend(order._prepare_invoice_lines())
+            new_move = orders._create_invoice(move_vals)
+
+            for order in orders:
+                order.write({'account_move': new_move.id, 'state': 'invoiced'})
```

Comment:
does it not work in batch? by just doing orders.write({'account_move': new_move.id, 'state': 'invoiced'}) ?